kanabi.ts

```diff
@@ -83,112 +92,143 @@ type _BuildArgs<
-    ExtractAbiFunctions<TAbi>['name'];
+  ExtractAbiFunctions<TAbi>['name'];
 
 export type ExtractAbiFunction<
   TAbi extends Abi,
   TFunctionName extends ExtractAbiFunctionNames<TAbi>
 > =
-  Extract<ExtractAbiFunctions<TAbi>, {name: TFunctionName}>;
+  Extract<ExtractAbiFunctions<TAbi>, { name: TFunctionName }>;
 
 export type ExtractAbiStructs<TAbi extends Abi> =
-    Extract<TAbi[number], {type: 'struct'}>;
+  Extract<TAbi[number], { type: 'struct' }>;
 
 export type ExtractAbiStructNames<TAbi extends Abi> =
-    ExtractAbiStructs<TAbi>['name'];
+  ExtractAbiStructs<TAbi>['name'];
 
 export type ExtractAbiStruct<
   TAbi extends Abi,
   TStructName extends ExtractAbiStructNames<TAbi>
 > =
-  Extract<ExtractAbiStructs<TAbi>, {name: TStructName}>;
+  Extract<ExtractAbiStructs<TAbi>, { name: TStructName }>;
+
+export type ExtractAbiEnums<TAbi extends Abi> =
+  Extract<TAbi[number], { type: 'enum' }>;
+
+export type ExtractAbiEnumNames<TAbi extends Abi> =
+  ExtractAbiEnums<TAbi>['name'];
+
+export type ExtractAbiEnum<
+  TAbi extends Abi,
+  TEnumName extends ExtractAbiEnumNames<TAbi>
+> =
+  Extract<ExtractAbiEnums<TAbi>, { name: TEnumName }>;
 
 // Question: why do we need TAbi extends Abi here, it's not used ?
 type PrimitiveTypeLookup<TAbi extends Abi> = {
   [_ in CairoFelt]: bigint
 } & {
-  [_ in CairoFunction]: number
-} & {
-  [_ in CairoTuple]: any  // TODO: implement tuples
-} & {
-  [_ in CairoInt]: number | bigint // Question: Why not just number ?
-} & {
-  [_ in CairoBigInt]: bigint
-} & {
-  [_ in CairoAddress]: bigint
-} & {
-  [_ in CairoVoid]: void
-}
+    [_ in CairoFunction]: number
+  } & {
+    [_ in CairoInt]: number | bigint // Question: Why not just number ?
+  } & {
+    [_ in CairoBigInt]: bigint
+  } & {
+    [_ in CairoAddress]: bigint
+  } & {
+    [_ in CairoVoid]: void
+  } & {
+    [_ in CairoBool]: boolean
+  }
 
 export type AbiTypeToPrimitiveType<TAbi extends Abi, TAbiType extends AbiType> =
-    PrimitiveTypeLookup<TAbi>[TAbiType];
-
+  PrimitiveTypeLookup<TAbi>[TAbiType];
 
 export type GenericTypeToPrimitiveType<TAbi extends Abi, G extends string> =
   G extends CairoOptionGeneric<infer T>
-    ? T extends AbiType
-      ? Option<AbiTypeToPrimitiveType<TAbi, T>>
-      : Option<GenericTypeToPrimitiveType<TAbi, T>>
-    : G extends CairoArrayGeneric<infer T>
-      ? T extends AbiType
-        ? AbiTypeToPrimitiveType<TAbi, T>[]
-        : GenericTypeToPrimitiveType<TAbi, T>[]
-    : unknown;
-
-
-// Question: AbiParameterToPrimitiveType<TAbi, {ty: "MissingStruct", name: 'x'}
-//           doesn't raise an error although there is no struct named
-//           'MissingStruct' defined in the ABI, is this the expected behavior?
-export type AbiParameterToPrimitiveType<
+  ? T extends AbiType
+  ? Option<AbiTypeToPrimitiveType<TAbi, T>>
+  : Option<StringToPrimitiveType<TAbi, T>>
+  : G extends CairoArrayGeneric<infer T>
+  ? T extends AbiType
+  ? AbiTypeToPrimitiveType<TAbi, T>[]
```

Comment:
Difficult to follow the branching since the absence of indentation.
Could you investigate how to add indentation for clarity ?